packages/core/src/components/pagination/pagination.tsx

```diff
@@ -49,30 +77,105 @@ export class Pagination implements ComponentInterface {
+            <a href={this.formatHref(pages[0])} aria-label="Vorige" onClick={e => currentPage && this.clickHandler(e, pages[currentPage - 2])}>
+              <dso-icon icon="chevron-left"></dso-icon>
+            </a>
+          </li>
+          {(pageRange).map(page => (
+            <>
+              {(pages.indexOf(page) === pages.length - 1 && pages.length > (this.elementPositions - 2)) &&
+                <>
+                { (
+                    (pages.indexOf(currentPage) === 0 && this.elementPositions === 6) ||
+                    (currentPage < pages[pages.length - 1] - 2 && this.elementPositions === 8) ||
+                    (currentPage < pages[pages.length - 1] - 4 && this.elementPositions === 10)
+                  ) &&
+                  <li>
+                    <span>...</span>
+                  </li>
+                }
+                </>
+              }
+
+              <li key={page} class={currentPage === page ? 'active' : undefined}>
+                {currentPage === page
+                  ? (
+                    <span aria-current="page">{page}</span>
+                  )
+                  : (
+                    <a href={this.formatHref(page)} onClick={e => this.clickHandler(e, page)}>{page}</a>
+                  )}
+              </li>
+
+              {(pages.indexOf(page) === 0 && pages.length > (this.elementPositions - 2)) &&
+                <>
+                  { (
+                      (pages.indexOf(currentPage) === pages.length - 1 && this.elementPositions === 6) ||
+                      (currentPage > pages[0] + 2 && this.elementPositions === 8) ||
+                      (currentPage > pages[0] + 4 && this.elementPositions === 10)
+                    )  &&
+                    <li>
+                      <span>...</span>
+                    </li>
+                  }
+                </>
+              }
+            </>
+          ))}
+          <li class={(currentPage >= pages[pages.length - 1] || currentPageOutOfBounds) ? 'dso-page-hidden' : undefined}>
+            <a href={this.formatHref(pages[pages.length - 1])} aria-label="Volgende" onClick={e => currentPage && this.clickHandler(e, pages[currentPage])}>
+              <dso-icon icon="chevron-right"></dso-icon>
+            </a>
           </li>
-        ))}
-        <li class={(currentPage >= pages[pages.length - 1] || currentPageOutOfBounds) ? 'dso-page-hidden' : undefined}>
-          <a href={this.formatHref(pages[pages.length - 1])} aria-label="Volgende" onClick={e => currentPage && this.clickHandler(e, pages[currentPage])}>
-            <dso-icon icon="chevron-right"></dso-icon>
-          </a>
-        </li>
-      </ul>
+        </ul>
+      </dso-responsive-element>
     );
   }
+
+  private getPageRange(pages: number[], currentPage: number, totalPages: number, elementPositions: number) {
+    return pages.length <= (elementPositions - 2) ? pages : pages.reduce<number[]>((prev, page) => {
```

Comment:
Beetje smaak maar ik zou de ternary vervangen met een `if`

Reply:
Daar kan ik mij in vinden. Maakt het wat leesbaarder.